env/io_posix.cc

```diff
@@ -409,13 +370,135 @@ size_t PosixHelper::GetUniqueIdFromFile(int fd, char* id, size_t max_size) {
   return static_cast<size_t>(rid - id);
 }
 #endif
+
+#ifdef OS_LINUX
+void LogicalBufferSizeCache::RefAndCacheLogicalBufferSize(
+    const std::set<std::pair<std::string, int>>& directories) {
+  std::map<int, size_t> fd_sizes;
```

Comment:
Some applications will `CreateColumnFamily` and `DropColumnFamily` frequently, but I assume the `cf_paths` for the `CreateColumnFamily` would not be too different, so I made the optimization here to only acquire write lock when there are new paths.